src/Gateway.php

```diff
@@ -0,0 +1,490 @@
+		 * Adyen checkout configuration.
+		 *
+		 * @link https://docs.adyen.com/checkout/drop-in-web
+		 * @link https://docs.adyen.com/checkout/components-web
+		 */
+		$configuration = [
+			'locale'                      => Util::get_payment_locale( $payment ),
+			'environment'                 => ( self::MODE_TEST === $payment->get_mode() ? 'test' : 'live' ),
+			'session'                     => (object) [
+				'id'          => $payment_session->get_id(),
+				'sessionData' => $payment_session->get_data(),
+			],
+			'clientKey'                   => $this->config->client_key,
+			'amount'                      => AmountTransformer::transform( $payment->get_total_amount() )->get_json(),
+			'paymentMethodsConfiguration' => $this->get_payment_methods_configuration( $payment ),
+		];
+
+		$configuration = (object) $configuration;
+
+		/**
+		 * Filters the Adyen checkout configuration.
+		 *
+		 * @param object $configuration Adyen checkout configuration.
+		 * @link https://docs.adyen.com/online-payments/drop-in-web#step-2-add-drop-in
+		 * @since 1.2.0 Added.
+		 */
+		$configuration = apply_filters( 'pronamic_pay_adyen_checkout_configuration', $configuration );
+
+		wp_localize_script(
+			'pronamic-pay-adyen-checkout',
+			'pronamicPayAdyenCheckout',
+			[
+				'configuration'    => $configuration,
+				'paymentReturnUrl' => $request->get_return_url(),
+			]
+		);
+
+		add_action( 'pronamic_pay_adyen_checkout_head', [ $this, 'checkout_head' ] );
+
+		Core_Util::no_cache();
+
+		require __DIR__ . '/../views/checkout-drop-in.php';
+
+		exit;
+	}
+
+	/**
+	 * Checkout head.
+	 *
+	 * @return void
+	 */
+	public function checkout_head() {
+		wp_print_styles( 'pronamic-pay-redirect' );
+
+		wp_print_scripts( 'pronamic-pay-adyen-checkout' );
+
+		wp_print_styles( 'pronamic-pay-adyen-checkout' );
+	}
+
+	/**
+	 * Create payment.
+	 *
+	 * @param Payment              $payment        Payment.
+	 * @param PaymentMethodDetails $payment_method Payment method.
+	 * @return PaymentResponse
+	 * @throws \InvalidArgumentException Throws exception on invalid amount.
+	 * @throws \Exception Throws exception if payment creation request fails.
+	 */
+	private function create_payment( Payment $payment, PaymentMethodDetails $payment_method ) {
+		$payment_id = (string) $payment->get_id();
+
+		$payment_request = new PaymentRequest(
+			AmountTransformer::transform( $payment->get_total_amount() ),
+			$this->config->get_merchant_account(),
+			$payment_id,
+			$this->get_payment_return_url( $payment_id ),
+			$payment_method
+		);
+
+		PaymentRequestHelper::complement( $payment, $payment_request, $this->config );
+
+		$payment_response = $this->client->create_payment( $payment_request );
+
+		/**
+		 * Store payment response for later requests to `/payments/details`.
+		 *
+		 * @link https://docs.adyen.com/api-explorer/#/PaymentSetupAndVerificationService/v51/payments/details
+		 */
+		$payment->set_meta( 'adyen_payment_response', $payment_response->get_json() );
+
```

Comment:
Do we still need this? I don't think we're using the payment response later anymore.

```suggestion
```